tracer/src/Datadog.Trace.Tools.Runner/CheckCommand.cs

```diff
@@ -0,0 +1,75 @@
+// <copyright file="CheckCommand.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.CommandLine;
+using System.CommandLine.Invocation;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using Spectre.Console;
+
+namespace Datadog.Trace.Tools.Runner
+{
+    internal class CheckCommand : Command
+    {
+        private readonly ApplicationContext _applicationContext;
+
+        public CheckCommand(ApplicationContext applicationContext)
+            : base("check")
+        {
+            _applicationContext = applicationContext;
+
+            AddArgument(Command);
+
+            this.SetHandler(Execute);
+        }
+
+        public Argument<string[]> Command { get; } = new("command", CommandLineHelpers.ParseArrayArgument, isDefault: true);
+
+        private void Execute(InvocationContext context)
+        {
+            var tracerHome = Utils.GetHomePath(_applicationContext.RunnerFolder);
+
+            // pick the right one depending on the platform
+            var ddDotnet = (platform: _applicationContext.Platform, arch: RuntimeInformation.OSArchitecture, musl: Utils.IsAlpine()) switch
+            {
+                (Platform.Windows, Architecture.X64, _) => Path.Combine(tracerHome, "win-x64", "dd-dotnet.exe"),
+                (Platform.Windows, Architecture.X86, _) => Path.Combine(tracerHome, "win-x64", "dd-dotnet.exe"),
```

Comment:
Should this folder be "win-x86" instead?

Reply:
That's a good question. dd-dotnet has no binary for win-x86 (not supported by NativeAOT), but nobody uses a x86 version of Windows nowadays sothe x64 version will work fine.